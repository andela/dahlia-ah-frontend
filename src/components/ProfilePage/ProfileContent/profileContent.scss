.profile-content {
  * {
    margin: 0;
    padding: 0;
  }
  display: flex;
  flex-direction: row;
  .name {
    font-size: 1.9em;
    text-transform: uppercase;
    padding: 0;
    margin: 0;
  }

  .bio {
    max-width: 70ch;
    min-width: 40ch;
    margin-top: 1rem;
  }

  .avatar-container {
    width: 200px;
    height: 200px;
    margin-right: 3rem;
    position: relative;

    .img {
      width: 100%;
      height: 100%;
    }

    .fas.fa-camera {
      visibility: hidden;
      font-size: 1.5em;
      color: rgb(0, 0, 0);
      padding: 3px;
      position: absolute;
      bottom: 0;
      right: 0;
    }

    &:hover,
    &:focus,
    &:active {
      .fas.fa-camera {
        visibility: visible;
      }
    }
  }

  .bio-container {
    display: flex;
    flex-flow: column nowrap;
  }

  .generic-btn,
  .generic-btn:focus,
  .generic-btn:active {
    padding: 0.5rem 1rem;
    margin: 10px 10px;
    border: 1px solid rgba(0, 0, 0, 0.6);
    align-self: center;
    font-size: 0.7rem;
    text-transform: uppercase;
    outline: none;
    background-color: white;
    cursor: pointer;
  }

  @media only screen and (max-width: 850px) {
    flex-direction: column;
    align-items: center;
    text-align: center;
    font-size: 90%;

    .avatar-container {
      width: 170px;
      height: 170px;
      margin: 2rem 0 1rem 0;
    }
  }

  @media only screen and (max-width: 540px) {
    font-size: 80%;
    .avatar-container {
      width: 150px;
      height: 150px;
    }
  }

  @media only screen and (max-width: 435px) {
    font-size: 70%;
    .avatar-container {
      width: 140px;
    }
  }
}
