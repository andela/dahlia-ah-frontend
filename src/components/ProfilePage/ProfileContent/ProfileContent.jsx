import React from 'react';
import PropTypes from 'prop-types';
import Stats from '../Stats/Stats';
import './profileContent.scss';

const ProfileContent = ({
  name, image, bio, followers, following, written,
}) => (
  <div className="profile-content">
    <div className="avatar-container">
      <img src={image} alt="avatar" />
    </div>
    <div className="bio-container">
      <h3 className="name">{name}</h3>
      <p className="bio">{bio}</p>
      <Stats following={following} followers={followers} written={written} />
      <button className="generic-btn" type="button">
          Edit Profile
      </button>
    </div>
  </div>
);

ProfileContent.propTypes = {
  name: PropTypes.string.isRequired,
  bio: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  following: PropTypes.number.isRequired,
  followers: PropTypes.number.isRequired,
  written: PropTypes.number.isRequired,
};

export default ProfileContent;
