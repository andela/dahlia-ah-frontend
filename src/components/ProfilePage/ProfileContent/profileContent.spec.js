import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProfileContent from './ProfileContent';

configure({ adapter: new Adapter() });

describe('Author Profile/Bio Component', () => {
  const props = {
    name: 'Some Name',
    bio: 'Bio text',
    image: 'https://some-image.com',
    following: 0,
    followers: 1,
    written: 2,
  };
  it('should render properly', () => {
    const wrapper = shallow(
      <ProfileContent
        name={props.name}
        bio={props.bio}
        image={props.image}
        followers={props.followers}
        following={props.following}
        written={props.written}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
