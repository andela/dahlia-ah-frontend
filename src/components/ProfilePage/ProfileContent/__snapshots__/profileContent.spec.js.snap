// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Profile/Bio Component should render properly 1`] = `ShallowWrapper {}`;
